obelix/idefix/services/register.go

```diff
@@ -0,0 +1,36 @@
+package services
+
+import (
+	"context"
+	"fmt"
+	"github.com/PtitLuca/go-dispatcher/dispatcher"
+	"idefix/operator"
+	"idefix/producer"
+	"idefix/redis"
+	"idefix/services/github"
+)
+
+func RegisterServices(ctx context.Context) (*dispatcher.Dispatcher, error) {
+	d := dispatcher.New()
+	kp, err := producer.New()
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to create kafka producer: %w", err)
+	}
+
+	rc := redis.NewClient(ctx)
+
+	githubClient := &github.Client{
+		Requester: nil,
+		Operator: &operator.IdefixOperator{
+			RC: rc,
+			KP: kp,
+		},
+	}
+
+	if err := d.Register("github", githubClient); err != nil {
```

Comment:
No need to redefine the error here, just re-assign it.

Reply:
0317d12d736f09313503b7b06cfcf219a4a65981